runtime/src/message_processor.rs

```diff
@@ -1257,11 +1260,13 @@ impl MessageProcessor {
                 )
                 .map_err(|err| TransactionError::InstructionError(instruction_index as u8, err));
             time.stop();
+            let post_remaining_units = compute_meter.borrow().get_remaining();
 
             let program_id = instruction.program_id(&message.account_keys);
-            let time_count = timings.per_program_timings.entry(*program_id).or_default();
-            time_count.accumulated_us += time.as_us();
-            time_count.count += 1;
+            let program_timing = timings.per_program_timings.entry(*program_id).or_default();
+            program_timing.accumulated_us += time.as_us();
+            program_timing.accumulated_units += pre_remaining_units - post_remaining_units;
+            program_timing.count += 1;
```

Comment:
Could we use saturating math here, just in case?

Reply:
Let's take that up as a separate issue, I'll keep the match consistent here